packages/ui/src/components/skeleton-list.tsx

```diff
@@ -0,0 +1,61 @@
+import { useEffect, useState } from 'react'
+
+import { Skeleton, StackedList } from '@/components'
+import { cn } from '@utils/cn'
+
+// Helper function to generate random percentage width within a range
+const getRandomPercentageWidth = (min: number, max: number) => `${Math.floor(Math.random() * (max - min + 1)) + min}%`
+
+// Helper function to generate random pixel width within a range
+const getRandomPixelWidth = (min: number, max: number) => `${Math.floor(Math.random() * (max - min + 1)) + min}px`
+
+interface SkeletonListProps {
+  className?: string
+}
+
+export const SkeletonList = ({ className }: SkeletonListProps) => {
+  const listItems = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+
+  // do a first render with visible=false
+  const [visible, setVisible] = useState(false)
+  // immediately re-render with visible=true to trigger the transition delay
+  useEffect(() => {
+    setVisible(true)
+  }, [])
+
+  return (
+    <div
+      className={cn(
+        'relative h-full w-full transition-opacity delay-500 duration-500 ease-in-out',
+        {
+          'opacity-0': !visible
+        },
+        className
+      )}
+    >
+      {listItems && listItems.length > 0 && (
+        <StackedList.Root>
+          {listItems.map((itm, itm_idx) => (
+            <StackedList.Item key={itm_idx} className="py-4" isLast={listItems.length - 1 === itm_idx}>
```

Comment:
Since we know everything in the `listItems` array is unique, we can avoid using the index.
```suggestion
          {listItems.map(item => (
            <StackedList.Item key={item} className="py-4" isLast={listItems.length === item}>
```